### ISSUE TITLE
adding backprop to feat

### ISSUE THREAD

**Issue by lacava** [lacava/feat#80]
# weights

- [x] add weights to continuous nodes so that the output is multiplied by the weights.
- [x] add a learning rate parameter $\nu$ (scales gradient update)
- [x] add a momentum parameter $\apha$
such that 

$ w_1 = w_0(1-\alpha) + \alpha \nu \Delta w_0 $

- [x] add a batch size parameter (fraction of data to use for weight updating)
- [x] add an iterations parameter to control number of iterations of weight updating

# algorithms

- [x] add derivative methods to nodes that update w 
- [x] add backprop() method that updates weights after evaluation

**Comment by lacava** [comment]
see #136
